core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/MethodConfigurationComponent.scala

```diff
@@ -96,35 +97,44 @@ trait MethodConfigurationComponent {
+    def save(workspaceContext: SlickWorkspaceContext, methodConfig: MethodConfiguration): ReadWriteAction[MethodConfiguration] = {
 
       uniqueResult[MethodConfigurationRecord](findByName(workspaceContext.workspaceId, methodConfig.namespace, methodConfig.name)) flatMap {
         case None =>
-          val configInsert = (methodConfigurationQuery returning methodConfigurationQuery.map(_.id) +=  marshalMethodConfig(workspaceContext.workspaceId, methodConfig))
-          configInsert flatMap { configId =>
-            saveMaps(configId)
+            val configInsert = (methodConfigurationQuery returning methodConfigurationQuery.map(_.id) +=  marshalMethodConfig(workspaceContext.workspaceId, methodConfig))
+            configInsert flatMap { configId =>
+              saveMaps(methodConfig, configId)
           }
         case Some(methodConfigRec) =>
-          workspaceQuery.updateLastModified(workspaceContext.workspaceId) andThen
-            findInputsByConfigId(methodConfigRec.id).delete andThen
-            findOutputsByConfigId(methodConfigRec.id).delete andThen
-            findPrereqsByConfigId(methodConfigRec.id).delete andThen
-            findById(methodConfigRec.id).map(rec => (rec.methodNamespace, rec.methodName, rec.methodVersion)).update(methodConfig.methodRepoMethod.methodNamespace, methodConfig.methodRepoMethod.methodName, methodConfig.methodRepoMethod.methodVersion) andThen
-            saveMaps(methodConfigRec.id) andThen
-            findByName(workspaceContext.workspaceId, methodConfig.namespace, methodConfig.name).map(_.rootEntityType).update(methodConfig.rootEntityType)
+          saveNewVersion(workspaceContext, methodConfigRec.id, methodConfigRec.name, methodConfigRec.methodConfigVersion, methodConfig)
       }
     } map { _ => methodConfig }
 
+    private def saveMaps(methodConfig: MethodConfiguration, configId: Long) = {
+      val prerequisites = methodConfig.prerequisites.map { case (key, value) => marshalConfigPrereq(configId, key, value) }
+      val inputs = methodConfig.inputs.map { case (key, value) => marshalConfigInput(configId, key, value) }
+      val outputs = methodConfig.outputs.map{ case (key, value) => marshalConfigOutput(configId, key, value) }
+
+      (methodConfigurationPrereqQuery ++= prerequisites) andThen
+        (methodConfigurationInputQuery ++= inputs) andThen
+        (methodConfigurationOutputQuery ++= outputs)
+    }
+
+    private def saveNewVersion(workspaceContext: SlickWorkspaceContext, oldId: Long, oldName: String, oldConfigVersion: Int, newMethodConfig: MethodConfiguration) = {
+      workspaceQuery.updateLastModified(workspaceContext.workspaceId) andThen
+        hideMethodConfigurationAction(oldId, oldName) andThen
+        (methodConfigurationQuery returning methodConfigurationQuery.map(_.id) += marshalMethodConfigNewVersion(workspaceContext.workspaceId, newMethodConfig, oldConfigVersion + 1)) flatMap { configId =>
```

Comment:
Can you instead do this:

```marshalMethodConfig(workspaceContext.workspaceId, newMethodConfig.copy(version=oldConfigVersion + 1) )```

That'd let you drop `marshalMethodConfigNewVersion` entirely.